griptape/events/event_bus.py

```diff
@@ -11,14 +12,20 @@
     from griptape.events import BaseEvent, EventListener
 
 
+_event_listeners: contextvars.ContextVar[list[EventListener]] = contextvars.ContextVar("event_listeners", default=[])
```

Comment:
Just curious, could this have been a field on the singleton?

Reply:
https://docs.python.org/3/library/contextvars.html#contextvars.ContextVar

> Context Variables should be created at the top module level and never in closures. [Context](https://docs.python.org/3/library/contextvars.html#contextvars.Context) objects hold strong references to context variables which prevents context variables from being properly garbage collected.

Reply:
I almost feel like we should have a comment about this... but I leave that decision to you